packages/web/src/views/Calendar/components/Event/Draft/Draft.tsx

```diff
@@ -40,21 +43,35 @@ export const Draft: FC<Props> = ({
   const { draft, isDragging, isResizing } = draftState;
 
   const isDrafting = useAppSelector(selectIsDrafting);
+  const isProcessing = useAppSelector(selectIsDraftProcessing);
 
   const formProps = useEventForm("grid");
+  const [preservedDraft, setPreservedDraft] = useState(draft);
 
-  if (isLoadingDOM || !draft || !isDrafting) return null;
+  useEffect(() => {
+    if (!isProcessing) {
+      setPreservedDraft(draft);
+    }
+  }, [isProcessing, draft]);
+
+  if (isLoadingDOM) return null;
+  if (!preservedDraft) return null;
+
+  if (preservedDraft && !isDrafting && !isProcessing) {
+    return null;
+  }
+  if (!preservedDraft && !draft) return null;
```

Comment:
This effect is probably what's getting us into trouble.  But I'm wondering if `draft` is changing too frequently for this copied `preservedDraft` to stay in sync